bee-common/src/packable/mod.rs

```diff
@@ -0,0 +1,178 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! A module that provides a `Packable` trait to serialize and deserialize types.
+
+mod packer;
+mod unpacker;
+
+pub use packer::{PackError, Packer};
+pub use unpacker::{UnpackError, Unpacker};
+
+/// A type that can be packed and unpacked.
+///
+/// Almost all basic sized types implement this trait. This trait can be derived using the
+/// `packable_derive` macro. If you need to implement this trait manually, use the provided
+/// implementations as a guide.
+pub trait Packable: Sized {
+    /// Pack this value into the given `Packer`.
+    fn pack<P: Packer>(&self, packer: &mut P) -> Result<(), P::Error>;
+    /// Unpack this value from the given `Unpacker`.
+    fn unpack<U: Unpacker>(unpacker: &mut U) -> Result<Self, U::Error>;
+}
+
+macro_rules! impl_packable_for_int {
+    ($ty:ty) => {
+        impl Packable for $ty {
+            fn pack<P: Packer>(&self, packer: &mut P) -> Result<(), P::Error> {
+                packer.pack_bytes(&self.to_le_bytes())
+            }
+
+            fn unpack<U: Unpacker>(unpacker: &mut U) -> Result<Self, U::Error> {
+                let bytes: &[u8; std::mem::size_of::<Self>()] = unpacker.unpack_exact_bytes()?;
+                Ok(Self::from_le_bytes(*bytes))
+            }
+        }
+    };
+}
+
+impl_packable_for_int!(u8);
+impl_packable_for_int!(u16);
+impl_packable_for_int!(u32);
+impl_packable_for_int!(u64);
+impl_packable_for_int!(u128);
```

Comment:
Can you conditionally exclude if u128 is not available please ? We had issues in the past because of that.
https://github.com/iotaledger/bee/blob/dev/bee-common/bee-common/src/packable.rs#L204
https://github.com/iotaledger/bee/blob/dev/bee-common/bee-common/build.rs